examples/platform_services/android/app/src/main/java/com/example/flutter/ExampleActivity.java

```diff
@@ -4,60 +4,43 @@
 import io.flutter.plugin.common.FlutterMethodChannel.Response;
 import io.flutter.plugin.common.MethodCall;
-import io.flutter.view.FlutterView;
 
 public class ExampleActivity extends FlutterActivity {
-    private FlutterView flutterView;
+  private static final String CHANNEL = "battery";
 
-    @Override
-    public void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        new FlutterMethodChannel(getFlutterView(), "geo").setMethodCallHandler(new MethodCallHandler() {
-            @Override
-            public void onMethodCall(MethodCall call, Response response) {
-                if (call.method.equals("getLocation")) {
-                    if (!(call.arguments instanceof String)) {
-                        throw new IllegalArgumentException("Invalid argument type, String expected");
-                    }
-                    getLocation((String) call.arguments, response);
-                } else {
-                    throw new IllegalArgumentException("Unknown method " + call.method);
-                }
-            }
-        });
-    }
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
 
-    private void getLocation(String provider, Response response) {
-        String locationProvider;
-        if (provider.equals("network")) {
-            locationProvider = LocationManager.NETWORK_PROVIDER;
-        } else if (provider.equals("gps")) {
-            locationProvider = LocationManager.GPS_PROVIDER;
-        } else {
-            throw new IllegalArgumentException("Unknown provider " + provider);
-        }
-        String permission = "android.permission.ACCESS_FINE_LOCATION";
-        if (checkCallingOrSelfPermission(permission) == PackageManager.PERMISSION_GRANTED) {
-            LocationManager locationManager = (LocationManager) getSystemService(Context.LOCATION_SERVICE);
-            Location location = locationManager.getLastKnownLocation(locationProvider);
-            if (location != null) {
-                response.success(new double[] { location.getLatitude(), location.getLongitude() });
+    new FlutterMethodChannel(getFlutterView(), CHANNEL).setMethodCallHandler(
+        new MethodCallHandler() {
+          @Override
+          public void onMethodCall(MethodCall call, Response response) {
+            if (call.method.equals("getBatteryLevel")) {
+              getBatteryLevel(response);
             } else {
-                response.error("unknown", "Location unknown", null);
+              throw new IllegalArgumentException("Unknown method " + call.method);
             }
-        } else {
-            response.error("permission", "Access denied", null);
-        }
+          }
+    });
+  }
+
+  private void getBatteryLevel(Response response) {
+    BatteryManager batteryManager = (BatteryManager) getSystemService(BATTERY_SERVICE);
+    if (VERSION.SDK_INT >= VERSION_CODES.LOLLIPOP) {
```

Comment:
Right now none of our examples work on API levels older than 19, since java.nio.charset.StandardCharsets is used at startup. Mikkel is looking in to fixing that. I'll leave this for now and take another look once the fix is there.